stock_planning/models/stock_planning.py

```diff
@@ -11,30 +11,15 @@ class StockPlanning(models.Model):
 
     _name = 'stock.planning'
     _description = 'Stock Planning'
-    _order = 'location, warehouse, product, scheduled_date'
 
-    @api.one
-    def _get_product_info_location(self):
-        self.qty_available = 0
-        self.virtual_available = 0
-        self.incoming_qty = 0
-        self.outgoing_qty = 0
-        if self.product and self.location:
-            prod = self.env['product.product'].with_context(
-                {'location': self.location.id}).browse(self.product.id)
-            self.qty_available = prod.qty_available
-            self.virtual_available = prod.virtual_available
-            self.incoming_qty = prod.incoming_qty
-            self.outgoing_qty = prod.outgoing_qty
+    company = fields.Many2one('res.company', 'Company')
+    warehouse = fields.Many2one('stock.warehouse', 'Warehouse')
+    location = fields.Many2one('stock.location', 'Location', translate=True)
+    from_date = fields.Date('From Date')
+    scheduled_date = fields.Date('Scheduled date')
 
-    @api.one
-    def _get_to_date(self):
-        self.move_incoming_to_date = 0
-        self.procurement_incoming_to_date = 0
-        self.outgoing_to_date = 0
-        self.scheduled_to_date = 0
+    def _calculate_move_incoming_to_date(self):
```

Comment:
Aquí, si es posible yo haría una función genérica,con parámetros para ejecutar el filtro en la query y que devuelva la cantidad en función de dichos parámetros. De esta forma evitamos repetir el mismo código para ver movimientos de entrada, o de salida... o incluso a futuro podriamos mirar internos. Lo haría así:

_calculate_move (self, company, location, product, location_origin, location_destiny, date_from, date_to, status_list)

 def _calculate_move_to_date (self, company, location, product, location_origin, location_destiny, date_from, date_to, status_list):
+        move_obj = self.env['stock.move']
+        move_qty = 0
+        cond = [('company_id', '=', company),
+                ('product_id', '=', product),
+                ('location_id', '=', location_origin),
+                ('location_dest_id', '=', location_destiny),
+                ('date', '>=', date_from),
+                ('date', '<=', date_to),
+                ('state', 'not in', status_.list]
+        moves = move_obj.search(cond)
+        if moves:
+            move_qty = sum(x.product_uom_qty for x in moves)
+        return move_qty

Y con esto tenemos un genérico al que podemos llamar desde cualquier sitio. Por supuesto poniendo la sintaxis correcta que esta seguro que no es.